tools/sof-kernel-log-check.sh

```diff
@@ -1,72 +1,120 @@
+ignore_str="$ignore_str"'|platform regulatory\.0: Direct firmware load for regulatory\.db failed with error -2'
+ignore_str="$ignore_str"'|cfg80211: failed to load regulatory\.db'
+ignore_str="$ignore_str"'|EXT4-fs \(nvme0n1p6\): re-mounted\. Opts: errors=remount-ro'
+ignore_str="$ignore_str"'|usb 2-3: Enable of device-initiated U1 failed\.'
+ignore_str="$ignore_str"'|usb 2-3: Enable of device-initiated U2 failed\.'
+ignore_str="$ignore_str"'|iwlwifi 0000:00:14\.3: Direct firmware load for iwlwifi-QuZ-a0-hr-b0-56\.ucode failed with error -2'
+ignore_str="$ignore_str"'|iwlwifi 0000:00:14\.3: Direct firmware load for iwlwifi-QuZ-a0-hr-b0-55\.ucode failed with error -2'
+ignore_str="$ignore_str"'|iwlwifi 0000:00:14\.3: Direct firmware load for iwlwifi-QuZ-a0-hr-b0-54\.ucode failed with error -2'
+ignore_str="$ignore_str"'|iwlwifi 0000:00:14\.3: Direct firmware load for iwlwifi-QuZ-a0-hr-b0-53\.ucode failed with error -2'
+ignore_str="$ignore_str"'|iwlwifi 0000:00:14\.3: Direct firmware load for iwlwifi-QuZ-a0-hr-b0-52\.ucode failed with error -2'
+ignore_str="$ignore_str"'|iwlwifi 0000:00:14\.3: Direct firmware load for iwlwifi-QuZ-a0-hr-b0-51\.ucode failed with error -2'
+ignore_str="$ignore_str"'|iwlwifi 0000:00:14\.3: Direct firmware load for iwlwifi-QuZ-a0-hr-b0-50\.ucode failed with error -2'
+ignore_str="$ignore_str"'|iwlwifi 0000:00:14\.3: Direct firmware load for iwlwifi-QuZ-a0-hr-b0-49\.ucode failed with error -2'
+ignore_str="$ignore_str"'|iwlwifi 0000:00:14\.3: Direct firmware load for iwl-debug-yoyo\.bin failed with error -2'
+ignore_str="$ignore_str"'|thermal thermal_zone.*: failed to read out thermal zone \(-61\)'
+
+# Dell CML-U laptop with SoundWire, issues reported by sof-test
+# BugLink: https://github.com/thesofproject/sof-test/issues/307
+ignore_str="$ignore_str"'|iwlwifi 0000:00:14\.3: Microcode SW error detected\. Restarting 0x0\.'
+ignore_str="$ignore_str"'|wlo1: authentication with f4:f5:e8:6b:45:bb timed out'
+
 [[ ! "$err_str" ]] && echo "Missing error keyword list" && exit 0
-# dmesg KB size buffer size
-#dmesg_config_define=$(awk -F '=' '/CONFIG_LOG_BUF_SHIFT/ {print $2;}' /boot/config-$(uname -r))
-#dmesg_buffer_size=$( echo $(( (1<<$dmesg_config_define) / 1024 )) )
-# kernel file log buffer size
-#kernel_buffer_size=$(du -k /var/log/kern.log |awk '{print $1;}')
-# now decide using which to catch the kernel log
-#[[ $kernel_buffer_size -lt $dmesg ]] && cmd="dmesg" || cmd="sed -n '$begin_line,\$p' /var/log/kern.log"
-
-# confirm begin_line is number, if it is not the number, direct using dmesg
-[[ "${begin_line//[0-9]/}" ]] && begin_line=0
-[[ "$begin_line" -eq 0 ]] && cmd="dmesg" || cmd="sed -n '$begin_line,\$p' /var/log/kern.log"
+
+# confirm begin_timestamp is valid, if it is not the number, search full log
+re='^[0-9]+$'
+if [[ $begin_timestamp =~ $re ]] ; then
+    echo "using all journal"
+    cmd='journalctl -k'
+else
+    cmd='journalctl -k --since=$begin_timestamp'
```

Comment:
I got the issue now. begin_timestamp has space inside. So we need to use quote for this args variable
```suggestion
    cmd="journalctl -k --since='$begin_timestamp'"
```

some simple test shared.
```
LASTTIME="2020-09-02 11:46:57.075394"
pxl@pxl-UP2:~$ cmd="journalctl -k --since='$LASTTIME'"
pxl@pxl-UP2:~$ eval $cmd
-- Logs begin at Wed 2019-12-11 11:26:27 CST, end at Wed 2020-09-02 11:51:20 CST. --
-- No entries --
pxl@pxl-UP2:~$ cmd="journalctl -k --since=$LASTTIME"
pxl@pxl-UP2:~$ eval $cmd
Failed to add match '11:46:57.075394': Invalid argument
```